mypy/checkstrformat.py

```diff
@@ -435,9 +435,9 @@ def perform_special_format_checks(
                 actual_type, "__str__"
             ):
                 self.msg.fail(
-                    'On Python 3 formatting "b\'abc\'" with "{}" '
-                    'produces "b\'abc\'", not "abc"; '
-                    'use "{!r}" if this is desired behavior',
+                    'If x = b\'abc\' then f"{x}" or "{}".format(x) produces "b\'abc\'", '
```

Comment:
Sadly, we have no way of telling whether it is a `.format` call or `fstring`. It would really help with the error message.

Reply:
Is that because we desugar f-strings into a CallExpr that looks just like a `str.format` call? Maybe we can stash some data in the CallExpr to indicate that it derived from an f-string.